From e62653fd08b2f572f94cf45166586452947789ef Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Mon, 5 May 2014 21:04:03 -0400
Subject: [block] nvme: Make I/O timeout a module parameter

Message-id: <1399323848-28227-8-git-send-email-dmilburn@redhat.com>
Patchwork-id: 79396
O-Subject: [RHEL7.1 PATCH BZ 1081734 07/12] NVMe: Make I/O timeout a module parameter
Bugzilla: 1081734
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Keith Busch <keith.busch@intel.com>

Increase the default timeout to 30 seconds to match SCSI.

Signed-off-by: Keith Busch <keith.busch@intel.com>
[use byte instead of ushort]
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>

(cherry picked from commit b355084a891985d4cd0ca23b1a83366af2c4232d)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/nvme-core.c b/drivers/block/nvme-core.c
index e5ccffe..c13fc67 100644
--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -50,6 +50,10 @@
 #define CQ_SIZE(depth)  (depth * sizeof(struct nvme_completion))
 #define ADMIN_TIMEOUT (60 * HZ)
 
+unsigned char io_timeout = 30;
+module_param(io_timeout, byte, 0644);
+MODULE_PARM_DESC(io_timeout, "timeout in seconds for I/O");
+
 static int nvme_major;
 module_param(nvme_major, int, 0);
 
diff --git a/include/linux/nvme.h b/include/linux/nvme.h
index 15d071e..1da0807 100644
--- a/include/linux/nvme.h
+++ b/include/linux/nvme.h
@@ -66,7 +66,8 @@ enum {
 
 #define NVME_VS(major, minor) (major << 16 | minor)
 
-#define NVME_IO_TIMEOUT (5 * HZ)
+extern unsigned char io_timeout;
+#define NVME_IO_TIMEOUT (io_timeout * HZ)
 
 /*
  * Represents an NVM Express device.  Each nvme_dev is a PCI function.
-- 
1.7.1